code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -381,9 +381,23 @@
 	if(isliving(target)) //will we ever even need to tranfer reagents to non-living mobs?
 		var/mob/living/L = target
 		perm = L.reagent_permeability()
-	multiplier *= perm
-
-	return trans_to_mob(target, amount*0.75, CHEM_TOUCH, multiplier, copy) + trans_to_mob(target, amount*0.25, CHEM_BREATHE, multiplier, copy)
+	
+	var/breath_amount = 0
+	var/taste_amount = 0
+	var/splash_amount = trans_to_turf(get_turf(target),(1-(perm*0.5))*amount)
+
+	if(prob(25))

```

Comment:
Jesus what the fuck is this pyramid of doom.

Early return or condense this logic somehow.